src/xml/node.cr

```diff
@@ -273,7 +274,14 @@ struct XML::Node
     if document? || text? || cdata? || fragment?
       raise XML::Error.new("Can't set name of XML #{type}", 0)
     end
-    LibXML.xmlNodeSetName(self, name.to_s)
+
+    name = name.to_s
+
+    if name.includes?('\0') || LibXML.xmlValidateNameValue(name) == 0
```

Comment:
Maybe we can call `name.check_no_null_byte` before this `if`

Reply:
I'd rather that we raised a single exception type for the case in which the name is invalid, instead of different ones based on how it's invalid.

Reply:
Sounds good. But for `\0` a message like "contains a null byte" might be more helpful.